### ISSUE TITLE
<mx:TextArea> tag may not function?

### ISSUE THREAD

**Issue by T-San001** [apache/royale-asjs#682]
When embedding an external mxml that uses <mx:TextArea>, the compiled output 
has nothing to display.

If comment out the following, the other portions is output.

<s:Label id="ActionTaken" x="10" y="77" text="Response:"/>
<com:NewTextArea id="responseContent" x="77" y="67" width="307" height="47" inputSet="true" editableSet="true" textAreaValue="Yes, sir."/>

[PSolve1C.xlsx](https://github.com/apache/royale-asjs/files/4063720/PSolve1C.xlsx)

**Comment by aharui** [comment]
Especially if nothing shows on screen, but also for most problems, please check the console output in the browser and report any exceptions found along with the stack information.

**Comment by T-San001** [comment]
Alex, thanks for your checking advice.

Here is the browser's console log.

Uncaught TypeError: Cannot set property 'editable' of undefined
    at HI.K.J5 (NewTextArea.mxml:32)
    at LY (MXMLDataInterpreter.js:199)
    at MP (MXMLDataInterpreter.js:127)
    at qG (MXMLDataInterpreter.js:271)
    at LY (MXMLDataInterpreter.js:244)
    at MP (MXMLDataInterpreter.js:127)
    at qG (MXMLDataInterpreter.js:271)
    at LY (MXMLDataInterpreter.js:244)
    at KY (MXMLDataInterpreter.js:62)
    at LY (MXMLDataInterpreter.js:186)

NewTextArea.mxml line 32 is:

				textArea.editable = value;

This is the part of the following:

			public function set editableSet(value:Boolean):void
			{
				_editableSet = value;
				textArea.editable = value;
				label.visible = value
			}

Any idea?

**Comment by aharui** [comment]
That implies that the TextArea's editable property has not yet been emulated.  You or other volunteers are welcome to contribute the emulation.  I think it has already been done for TextInput and could be copied.

**Comment by T-San001** [comment]
Alex, thanks. I take this as my assignment.

**Comment by T-San001** [comment]
Alex, editable is already defined in TextArea.as, which is different a bit than the one for TextInput.as.

Since the same mxml source (except for the namespaces) works in FlashBuider, editable in TextArea.as may have a problem?

==================================================

mx.controls.TextArea

    public function set editable(value:Boolean):void
    {
        if (value == _editable)
            return;

        _editable = value;
        editableChanged = true;
		
		COMPILE::JS {
			element["readOnly"] = !value;
		}

        //invalidateProperties();

        dispatchEvent(new Event("editableChanged"));
    }

=======================================
mx.controls.TextInput

    public function set editable(value:Boolean):void
    {
        _editable = value;
	COMPILE::JS
	{
		if(value == false) {
			(element as HTMLInputElement).readOnly = true;
		}
		else {
			 (element as HTMLInputElement).readOnly = value;
		}
	}
	
     /*   if (value == _editable)
            return;

        _editable = value;
        editableChanged = true;

        invalidateProperties();

        dispatchEvent(new Event("editableChanged")); */
    }

**Comment by aharui** [comment]
My apologies.  I did not look carefully at the console output.  Please debug in the js-debug version first.  That helps determine if it is a minification problem or not.

It will also be easier to debug as you should get useful line numbers.  It may be that your custom class doesn't subclass TextArea, but wraps one and it hasn't been created yet.  If you can show us code that causes the error or the entire NewTextArea.as, we can hopefully help you more.

**Comment by T-San001** [comment]
Alex, no problem.  

Please get the PSolve1c.xlsx to find the source code.  It is the attachment.

I checked the debug version.  The following is the console log.

=============================================================
Language.js:242 invalidateParentSizeAndDisplayList not implemented

Uncaught TypeError: Cannot set property 'editable' of undefined
    at jp.co.tak.view.usercomponent.NewTextArea.set__editableSet (NewTextArea.js:188)
    at Function.org.apache.royale.utils.MXMLDataInterpreter.initializeStrandBasedObject (MXMLDataInterpreter.js:199)
    at Function.org.apache.royale.utils.MXMLDataInterpreter.generateMXMLArray (MXMLDataInterpreter.js:127)
    at Function.org.apache.royale.utils.MXMLDataInterpreter.generateMXMLInstances (MXMLDataInterpreter.js:271)
    at Function.org.apache.royale.utils.MXMLDataInterpreter.initializeStrandBasedObject (MXMLDataInterpreter.js:244)
    at Function.org.apache.royale.utils.MXMLDataInterpreter.generateMXMLArray (MXMLDataInterpreter.js:127)
    at Function.org.apache.royale.utils.MXMLDataInterpreter.generateMXMLInstances (MXMLDataInterpreter.js:271)
    at Function.org.apache.royale.utils.MXMLDataInterpreter.initializeStrandBasedObject (MXMLDataInterpreter.js:244)
    at Function.org.apache.royale.utils.MXMLDataInterpreter.generateMXMLObject (MXMLDataInterpreter.js:62)
    at Function.org.apache.royale.utils.MXMLDataInterpreter.initializeStrandBasedObject (MXMLDataInterpreter.js:186)

**Comment by aharui** [comment]
I am unable to see all of the code using Excel.  Is there a reason you chose Excel?  You should be able to attach the source file to this issue.  Or post a link to it.

From what I saw it It looks like a lifecycle issue.  NewTextArea must create an internal TextArea at some point, but it has not been created at the time the MXMLDataInterpreter sets the editable property.

**Comment by T-San001** [comment]
Alex, sorry.  I attach the files.  Appreciate your advice on how to change.  Changed the file type to txt.

[NewTextArea.txt](https://github.com/apache/royale-asjs/files/4069875/NewTextArea.txt)
[PSolv1C.txt](https://github.com/apache/royale-asjs/files/4069877/PSolv1C.txt)

**Comment by aharui** [comment]
In Royale, child components are created later.  So properties that directly access child components will have problems.  I'm going to think about whether Royale can create child components earlier.  I think we don't want to for performance and CSS reasons.  So at least temporarily, you will have to adjust your code to store the property values and apply them later, probably in an override of addedToParent().

I'm kind of surprised your code worked in Flex.  I didn't think child component creation was early enough to be in time.  I will have to look into this more when I find a moment.

**Comment by T-San001** [comment]
Alex, I found the cause of the problem.  For the simplified test case, I used editableSet = "true".  But the original coding is actually data binding. 

By changing the data binding to the real value, the property passing to the child component seems to have a problem as you have explained.  I changed the parent mxml as follows and the problem went away.  Thanks for your taking time.

`			[Bindable]	
			public var responseContentFlag:Boolean = true;

                       for <com:NewTextArea>, editableSet="{responseContentFlag}"
`
